source/server/connection_handler_impl.cc

```diff
@@ -52,6 +38,7 @@ void ConnectionHandlerImpl::addListener(absl::optional<uint64_t> overridden_list
       }
       NOT_REACHED_GCOVR_EXCL_LINE;
     }
+    // TODO(lambdai): Remove the dependency of ActiveTcpListener.
```

Comment:
@ggreenway This is the why active_tcp_listener.h is required at this PR.
The next PR will erase the dependency.

Reply:
That's fine, but the way to pull that in is via deps, not hdrs. I don't see a good reason to do it via hdrs, even temporarily.

Reply:
There was a circle dependency if I don't use hdrs :(

Active_tcp_listener target depends on connection_handler_impl target: require `ConnectionImplHandler:ActiveListenerImplBase` in connection_handler_impl.h.

I can add the active_tcp_listener_header target to avoid this `hdrs` issue, like below. Hold on

active_tcp_listener_header: connection_handler_impl.h, active_tcp_listener.h
connection_handler_impl :  connection_handler_impl.cc, active_tcp_listener_header
active_tcp_listener :  active_tcp_listener.cc, active_tcp_listener_header

In active_tcp_listener.h
``` 
class ActiveTcpListener : public Network::TcpListenerCallbacks,
                          public ConnectionHandlerImpl::ActiveListenerImplBase,
```

Reply:
Oh, I see, I didn't understand that there was a circular dependency. Can you add a comment in the BUILD file explaining that, along with a TODO to fix?

Reply:
Add the explanation and TODO in BUILD